Editor/AGS.Types/Settings.cs

```diff
@@ -159,7 +159,30 @@ public void GenerateNewGameID()
         public string GameFileName
         {
             get { return _gameFileName; }
-            set { _gameFileName = value; }
+            set
+            {
+                // an approximation of the Editors validation function (without relying on Linq)

```

Comment:
Because I don't know what third party applications are consuming AGS.Types and whether they are using platforms where LINQ is available/usable. So to me it seems safer to not add a new dependency.

Reply:
Sorry I don't get this. LINQ is not a 3rd party framework, it's bundled with the .NET framework, there's no new dependencies added.

Reply:
I didn't say it was a 3rd part framework, I sent it might not be available or usable, because I don't know in advance that someone is actually using .NET Framework if they are using this project.

Reply:
What? Then what are they using? This is .NET code.

Reply:
I am concerned with the implementation. If someone builds game project tools with this in Mono, are you going to vouch that the LINQ implementation they are using is 100% correct?

I think perhaps you are just reading too much into my comment. If we forget any other project for the moment, if I just showed you this:
```
foreach (char c in value)
{
    if (c > 127)
    {
        throw new ArgumentException("Game file name should only contain letters and numbers");
    }
}
```
...would you ask me to introduce LINQ and use it instead, when nothing else in the project uses LINQ? If the answer is yes, I'm happy to change it.

Reply:
> are you going to vouch that the LINQ implementation they are using is 100% correct?

Yeah I will actually. LINQ has become a ubiquitous part of the .NET framework and has been so for over 10 years. There's no C# code today that is written without having at least a tiny part of LINQ in it, irregardless of whether they're compiling against the .NET framework, Mono, .NET core, .NET Standard or whatever open source fork of the runtime that might exist. That's a claim I can make with great confidence.

It also doesn't make sense to discriminate LINQ specifically, it's just as much a part of the .NET framework as anything else, should we also be cautious of System.IO.Path, or System.IO.File in case they were improperly implemented? Why was LINQ specifically selected?

> ...would you ask me to introduce LINQ and use it instead, 

I can tell you that I definitely would because I do so in every other project I participate in, being professionally or hobby, and so does many other C# devs I collab with. That being said, we rarely do so as a demand, but a friendly suggestion on writing nicer code. In this case I was actually considering suggesting it too FYI, but here it is a tiny improvmenet at best, and I noticed your comment on the top.

> when nothing else in the project uses LINQ?

The only reason that is, is because LINQ wasn't available when most of the code was written. That is not a very good reason to not use this fine tool now that we finally have it.


I'm not trying to be a hardass here, my biggest concern is that someone else will see your comment and they will feel like they should avoid LINQ because of it. Also I'm confident your concerns are completely unwarranted.

Reply:
> Why was LINQ specifically selected?

Because I was copying and pasting LINQ code into a project where the current dependencies wouldn't allow it to run. My comment was to explain why the code is different, not because I think LINQ is bad. Breaking things is bad, which is why I took the approach which didn't introduce new dependencies.

But, as you are saying it is 100% safe, then I'll switch it to use LINQ.

Reply:
Great. Sorry for being stubborn.